### ISSUE TITLE
primitives and spec missing on website

### ISSUE THREAD

**Issue by iacore** [thesephist/oak#7]
Hello @thesephist 

I can't find the primitives like `input()` on oaklang.org/.

I also have some other questions
- is there `read` and `write` from lisp
- how to use Oak inside an application as scripting language

**Comment by SpcFORK** [comment]
You can use Oak inside of another language easily if you and the language have Shell access and can keep doing so.
Just install Oak, and then use 'oak' to enter the repl.

**Comment by SpcFORK** [comment]
> I can't find the primitives like `input()` on oaklang.org/.

You could probably find some clues in the 'env.go' file starting at oakinput on line 577.

**Comment by iacore** [comment]
I end up finding the built-in functions in [docs/spec.md](https://github.com/thesephist/oak/blob/21db2d8394f4ad64dfa1198e24bbf516285499fc/docs/spec.md)

**Comment by SpcFORK** [comment]
This is true, but there are a few variables that aren't listed there inside the env

![image](https://github.com/thesephist/oak/assets/126259962/6ab248e8-0ded-4ef6-8fe7-171d60f2cf3b)

As you can see, these variables are usable within the latest Oak.
They just aren't in the old `spec.md`.

**Comment by iacore** [comment]
lib -> import
gc -> collect garbage
mem -> show memory usage
lib__oak_qm -> ?

**Comment by SpcFORK** [comment]
>  lib__oak_qm -> ?

lib__oak_qm -> import exists in memory?

**Comment by iacore** [comment]
> lib__oak_qm -> import exists in memory?

i can't find it in the source code

**Comment by SpcFORK** [comment]
> > lib__oak_qm -> import exists in memory?
> 
> i can't find it in the source code

Well, you can use it to get a true or false if a lib has been imported, e.g. `___runtime_lib?('std')`

**Comment by iacore** [comment]
I also found `___runtime_proc()`, which returns the current pid and `oak` excutable name.

```
	c.LoadFunc("___runtime_lib", c.rtLib)
	c.LoadFunc("___runtime_lib?", c.rtIsLib)
	c.LoadFunc("___runtime_gc", c.rtGC)
	c.LoadFunc("___runtime_mem", c.rtMem)
	c.LoadFunc("___runtime_proc", c.rtProc)
```

Oak is a good language. Will the author ever come back to update the official website?

**Comment by SpcFORK** [comment]
Not sure, but I've been using it with Kaboom for my games.

It's pretty good when you build to the web since you can interop between JS and Oak almost seamlessly.

**Comment by iacore** [comment]
> It's pretty good when you build to the web since you can interop between JS and Oak almost seamlessly.

How do I do that interop? Do I `print` the object to string and send it?

**Comment by SpcFORK** [comment]
> > It's pretty good when you build to the web since you can interop between JS and Oak almost seamlessly.
> 
> How do I do that interop? Do I `print` the object to string and send it?

No, you have to be building with `--web`

**Comment by iacore** [comment]
I misunderstood. I thought you were mentioning interop between Oak code in browser and Oak code on server.

I want to add basic features such as `chdir` to Oak. Can I submit the patch to your fork?
